nemo/collections/asr/parts/submodules/ctc_greedy_decoding.py

```diff
@@ -203,6 +218,94 @@ def forward(
 
         return (packed_result,)
 
+    @torch.no_grad()
+    def _greedy_decode_logprobs_batched(self, x: torch.Tensor, out_len: torch.Tensor):
+        # x: [B, T, D]
+        # out_len: [B]
+
+        batch_size = x.shape[0]
+
+        predictions = x
```

Comment:
Pleasr add detailed comments inline per stage of operation similar to how CTC and RNNT decoding algo is documented.